docs/flagd_kube_proxy.md

```diff
@@ -0,0 +1,53 @@
+# flagd-kube-proxy
```

Comment:
Calling this kube-proxy is so confusing as kube-proxy is a kubernetes component 

https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/

Reply:
Given that what we really do here is to proxy CRD access, another alternative is to call this `flagd-crd-proxy` 🤔

Reply:
@AlexsJones are you happy with `flagd-crd-proxy`, I can make these changes tomorrow

Reply:
What do you think about calling this `flagd-proxy`? While it's intended for Kubernetes, it's not a hard requirement. That should address the concern raised by Alex and provide flexibility if the proxy is required for other deployment patterns.

Reply:
I strongly agree with switching to `flagd-proxy`, it decouples us nicely from any underlying flag sources

Reply:
I'm OK with `flagd-proxy`.